### ISSUE TITLE
Hover cards are inaccessible in Teams high contrast theme

### ISSUE THREAD

**Issue by ling1726** [microsoft/fluentui#31764]
![image](https://github.com/microsoft/fluentui/assets/20744592/dc2b3a84-6cb6-417c-8c8d-e119b1906792)

There is insufficient contrast between white text and cyan background on hover - This is reproducible on the docsite

**Comment by ling1726** [comment]
Assigning @marcosmoura  as owner of the card component to triage
